### PR TITLE
refactor(remediate): RHICOMPL-993 consume remediation issue ID from GQL

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: src/PresentationalComponents/ComplianceRemediationButton/ComplianceRemediationButton.js [hunk 1]
```diff
@@ -50,18 +48,12 @@ class ComplianceRemediationButton extends React.Component {
...
       )
       .map((rule) => {
-        const profile = this.ruleProfile(rule, system);
-        return this.formatRule(
-          rule,
-          profile.refId,
-          system.id,
-          profile.osMajorVersion
-        );
+        return this.formatRule(rule, system.id);
```
Comment (vkrizan) [inline src/PresentationalComponents/ComplianceRemediationButton/ComplianceRemediationButton.js:56]: Nit: Can use implicit return:
```
.map(rule => this.formatRule(rule, system.id))
```

File: src/PresentationalComponents/ComplianceRemediationButton/ComplianceRemediationButton.js [hunk 2]
```diff
@@ -98,15 +90,15 @@ class ComplianceRemediationButton extends React.Component {
     return data.issues.length > 0 && data.systems.length > 0;
   };
 
-  remediationAvailable = () => {
+  remediationIssueId = () => {
```
Comment (vkrizan) [inline src/PresentationalComponents/ComplianceRemediationButton/ComplianceRemediationButton.js:93]: The name should stay `remediationAvailable` here.

File: src/PresentationalComponents/RulesTable/Cells.js [hunk 3]
```diff
@@ -76,7 +76,7 @@ export const Passed = ({ compliant }) =>
   );
 Passed.propTypes = ruleProps;
 
-export const RemediationColumnCell = ({ remediationAvailable }) => (
-  <RemediationCell hasPlaybook={remediationAvailable} />
+export const RemediationColumnCell = ({ remediationIssueId }) => (
+  <RemediationCell hasPlaybook={remediationIssueId} />
```
Comment (vkrizan) [inline src/PresentationalComponents/RulesTable/Cells.js:80]: Nit: hasPlaybook likely expects boolean.

**Review by vkrizan — APPROVED** [review]
Noice.

**Review by bastilian — APPROVED** [review]
LGTM! Thank you @skateman for taking care of this!

**Pull request merged**